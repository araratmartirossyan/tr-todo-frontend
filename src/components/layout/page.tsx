import { FC } from 'react'
import styled from 'styled-components'

export const StyledPage = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`

export const Page: FC = ({ children }) => <StyledPage>{children}</StyledPage>
