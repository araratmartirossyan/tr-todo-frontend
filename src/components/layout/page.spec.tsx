import { shallow, ShallowWrapper } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Page } from './page'

describe('Page test', () => {
  let component: ShallowWrapper
  beforeEach(() => {
    component = shallow(
      <Page>
        <h2 data-test-id="test-item">Hello world</h2>
      </Page>
    )
  })

  it('should render component', () => {
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render children element', () => {
    const childElement = component.find('[data-test-id="test-item"]')
    expect(childElement.exists()).toBe(true)
  })
})
