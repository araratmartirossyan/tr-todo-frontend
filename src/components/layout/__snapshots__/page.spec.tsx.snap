// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page test should render component 1`] = `
<styled.div>
  <h2
    data-test-id="test-item"
  >
    Hello world
  </h2>
</styled.div>
`;
